opentelekomcloud/config.go

```diff
@@ -198,6 +211,90 @@ func (c *Config) LoadAndValidate() error {
 	return nil
 }
 
+func newhwClient(c *Config) error {
+	ao := golangsdk.AuthOptions{
+		DomainID:         c.DomainID,
+		DomainName:       c.DomainName,
+		IdentityEndpoint: c.IdentityEndpoint,
+		Password:         c.Password,
+		TenantID:         c.TenantID,
+		TenantName:       c.TenantName,
+		TokenID:          c.Token,
+		Username:         c.Username,
+		UserID:           c.UserID,
+	}
+
+	client, err := huaweisdk.NewClient(ao.IdentityEndpoint)
+	if err != nil {
+		return err
+	}
+
+	// Set UserAgent
+	client.UserAgent.Prepend(terraform.UserAgentString())
+
+	config := &tls.Config{}
+	if c.CACertFile != "" {
+		caCert, _, err := pathorcontents.Read(c.CACertFile)
+		if err != nil {
+			return fmt.Errorf("Error reading CA Cert: %s", err)
+		}
+
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM([]byte(caCert))
+		config.RootCAs = caCertPool
+	}
+
+	if c.Insecure {
+		config.InsecureSkipVerify = true
+	}
+
+	if c.ClientCertFile != "" && c.ClientKeyFile != "" {
+		clientCert, _, err := pathorcontents.Read(c.ClientCertFile)
```

Comment:
You may want to move some of this common initialization in to a shared func. There is a good amount of repetition between the two initialization functions. The azurerm provider [has an example](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/config.go) of how multiple client initialization has had a lot of the repetition cleaned up.

Reply:
Good suggestion! Thanks!